java/ql/src/semmle/code/java/security/QueryInjection.qll

```diff
@@ -1,8 +1,7 @@
 /** Provides classes to reason about database query language injection vulnerabilities. */
 
-import java
```

Comment:
Don't remove `import java`. It's important to always have this as the first import, as this ensures consistent type discovery order in the optimiser (there's a very subtle issue where permuted import order can lead to different tiebreaker decisions, which in turn can lead to subtly different compilation for different queries, which causes re-computation due to cache misses and thereby loss of performance).

Reply:
I'll add this to my notes.